uefi/src/proto/device_path/mod.rs

```diff
@@ -326,6 +356,75 @@ impl DevicePath {
             stop_condition: StopCondition::EndEntireNode,
         }
     }
+
+    #[cfg(feature = "alloc")]
+    pub fn to_text_by_node(
+        &self,
+        bs: &BootServices,
```

Comment:
This is another thing I don't like about this crate: having to pass st/bs to pretty much everything. And that then sometimes complicating lifetimes.

Honestly, I wish this crate would make st/bs services a module with free-form functions (that access the global pointer stored somewhere) instead of making the functions impl on the struct that has to be passed around.